packages/@aws-cdk/aws-stepfunctions/lib/states/map.ts

```diff
@@ -150,10 +150,30 @@ export class Map extends State implements INextable {
      * Validate this state
      */
     protected validate(): string[] {
-        if (!!this.iterator) {
-            return ['Map state must have a non-empty iterator'];
+        const validateMaxConcurrency = () => {
+            const maxConcurrency = this.maxConcurrency;
+            if (maxConcurrency === undefined) {
+                return;
+            }
+
+            const isFloat = Math.floor(maxConcurrency) !== maxConcurrency;
+
+            const isNotPositiveInteger = maxConcurrency < 0 || maxConcurrency > Number.MAX_SAFE_INTEGER;
+
+            if (isFloat || isNotPositiveInteger) {
+                errors.push('maxConcurrency has to be a positive integer');
+            }
+        };
+
+        const errors: string[] = [];
+
+        if (this.iteration === undefined) {
```

Comment:
now Map has a method named iterator (like the property of the state in the doc) and the property on the parent State object is named iteration. I think that the method or the property should be renamed

Reply:
There's something to be said for that. `iteration` is `protected` though, so will be invisible to regular consumers. What situation are you concerned about?

Reply:
Oh I see, this bug. Yes, that's unfortunate, and thanks for the fix. Given the test, I think there's no need to rename now.